src/Symfony/Component/Scheduler/RecurringMessage.php

```diff
@@ -37,35 +40,53 @@ private function __construct(
      *  * A relative date format as supported by \DateInterval;
      *  * A \DateInterval instance.
      *
+     * @param MessageProviderInterface|object $message A message provider that yields messages or a static message that will be dispatched on every trigger
+     *
      * @see https://en.wikipedia.org/wiki/ISO_8601#Durations
      * @see https://php.net/datetime.formats.relative
      */
     public static function every(string|int|\DateInterval $frequency, object $message, string|\DateTimeImmutable $from = new \DateTimeImmutable(), string|\DateTimeImmutable $until = new \DateTimeImmutable('3000-01-01')): self
     {
-        return new self(new PeriodicalTrigger($frequency, $from, $until), $message);
+        return self::trigger(new PeriodicalTrigger($frequency, $from, $until), $message);
     }
 
+    /**
+     * @param MessageProviderInterface|object $message A message provider that yields messages or a static message that will be dispatched on every trigger
+     */
     public static function cron(string $expression, object $message, \DateTimeZone|string $timezone = null): self
     {
         if (!str_contains($expression, '#')) {
-            return new self(CronExpressionTrigger::fromSpec($expression, null, $timezone), $message);
+            return self::trigger(CronExpressionTrigger::fromSpec($expression, null, $timezone), $message);
         }
 
         if (!$message instanceof \Stringable) {
             throw new InvalidArgumentException('A message must be stringable to use "hashed" cron expressions.');
         }
 
-        return new self(CronExpressionTrigger::fromSpec($expression, (string) $message, $timezone), $message);
+        return self::trigger(CronExpressionTrigger::fromSpec($expression, (string) $message, $timezone), $message);
     }
 
+    /**
+     * @param MessageProviderInterface|object $message A message provider that yields messages or a static message that will be dispatched on every trigger
+     */
     public static function trigger(TriggerInterface $trigger, object $message): self
     {
-        return new self($trigger, $message);
+        if ($message instanceof MessageProviderInterface) {
+            return new self($trigger, $message);
+        }
+
+        try {
+            $description = $message instanceof \Stringable ? (string) $message : serialize($message);
+        } catch (\Exception) {
+            $description = $message::class;
+        }
+
+        return new self($trigger, new CallbackMessageProvider(static fn (): array => [$message], $description));
```

Comment:
instead of creating a closure here, maybe we should have a `FixedMessageProvider` taking an array of objects, or something like that.